radix-engine-interface/src/blueprints/epoch_manager/invocations.rs

```diff
@@ -50,9 +51,68 @@ pub const EPOCH_MANAGER_NEXT_ROUND_IDENT: &str = "next_round";
 
 #[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerNextRoundInput {
+    /// Current round number.
+    /// Please note that in case of liveness breaks, this number may be different than previous
+    /// reported `round + 1`. Such gaps are considered "round leader's fault" and are penalized
+    /// on emission, according to leader reliability statistics (see `LeaderProposalMissHistory`).
     pub round: u64,
```

Comment:
To ensure uniqueness of round transactions (which we probably don't actually need any more, given we've changed how the DB is indexed), we include an epoch in their payload on the node. I wonder if we should include an epoch here for sanity checking? Although I guess the fact that the `LeaderProposalHistory` length has to align with the difference in `round` is quite a good sanity check anyway.

Reply:
Separately, we want to move the clock to the EpochManager - so it might be worth sticking the `proposed_timestamp` on here (possibly none on fallback vertices) - @talekhinezh - do you think now's a good time to do that, given Jakub is touching round changes?

Reply:
> I wonder if we should include an epoch here for sanity checking?

I am closer to removing the round number :p

> Separately, we want to move the clock to the EpochManager

I am not sure what is the motivation for this, but even if very valid, then definitely a separate refactor 👍